### PR TITLE
Got downloading in binary mode working

### PR DESCRIPTION
**Pull request by samuel-massinon**
https://github.com/invenia/FTPClient.jl/issues/20

### CODE REVIEW

**Comment by samuel-massinon** [comment]
@omus Want to take a look at this?

**Comment by omus** [comment]
I'll take a look tomorrow

**Comment by omus** [comment]
Should `binary_mode` default to true? I expect that both binary files and ASCII files would download correctly in that mode.

**Comment by omus** [comment]
Maybe I'm missing something but shouldn't we have tests for retrieving ASCII files as binary and binary files as ASCII?

**Comment by samuel-massinon** [comment]
I put it false because that's what FTPClient.jl has been doing so far. Though it should be an easy change and not make a difference.
Most FTPClients look at the file they are downloaing and pick either binary or ascii, and if they aren't sure, they'll default to binary.
An example of when to use ascii is when downloading a windows text file to a unix system, the FTPClient will convert `\r\n` to `\n` for you, (and maybe more).

I do have some tests for downloading a binary file in ascii mode and showing it doesn't work. I was think of adding the other way (and more tests) next week.

**Comment by iamed2** [comment]
It's okay to make breaking changes as long as you respect semver IMO

**Comment by omus** [comment]
I am also ok with introducing a breaking change here. We can always do the type autodetection before the next release.

I'm not entirely sure we should be modifying the `\r\n` to `\n`...

**Comment by samuel-massinon** [comment]
We aren't changing `\r\n` to `\n`, it's LIBCurl and the black magic behind it. I think it's a good idea to default to binary (and maybe even have a smart enough FTPClient to determine what to download in, but not for this PR).

**Comment by omus** [comment]
Agreed. To be clear we're going to change the default to binary?

**Comment by samuel-massinon** [comment]
Yes.

**Comment by omus** [comment]
:+1:

**Comment by samuel-massinon** [comment]
@omus Ready to review now.

File: README.md [hunk 1]
```diff
@@ -75,11 +76,12 @@ ftp_close_connection(ctxt::ConnContext)
     - `ssl`: use FTPS, default is false
     - `verify_peer`: verify authenticity of peer's certificate, default is true
     - `active_mode`: use active mode to establish data connection, default is false
+    - `binary_mode`: used to tell the client to download files in binary mode, default is true
 
 
 #### FTPObject functions
 ```julia
-FTP(;host="", block=true, implt=false, ssl=false, ver_peer=true, act_mode=false, user="", pswd="")
+FTP(;host="", block=true, implt=false, ssl=false, ver_peer=true, act_mode=false, user="", pswd="", binary_mode=false)
```
Comment (omus) [inline README.md]: Should this now be set to true?

File: src/FTPObject.jl [hunk 2]
```diff
@@ -268,6 +270,7 @@ Set the transfer mode to binary.
 """ ->
 function binary(ftp::FTP)
 
+    ftp.ctxt.options.binary_mode = true
```
Comment (omus) [inline src/FTPObject.jl:273]: We should consider renaming these to `binary!` In the future

File: README.md (hunk 1 above)
Comment (samuel-massinon) [inline README.md]: Yes, fixed.

File: src/FTPObject.jl (hunk 2 above)
Comment (samuel-massinon) [inline src/FTPObject.jl:273]: What are the benefits of using `binary!`? Is it to following a standard?

Comment (omus) [inline src/FTPObject.jl:273]: Function names ending with an `!` indicate that the function is modifying the arguments. Mainly it's just a [nice convention](http://docs.julialang.org/en/release-0.4/stdlib/base/#introduction) to indictate that the function will be modifying a parameter.

**Comment by omus** [comment]
LGTM

**Pull request merged**